packaging/installer/methods/macos.md

```diff
@@ -1,42 +1,74 @@
 
-Netdata on macOS still has limited charts, but external plugins do work.
+Netdata works on macOS, albeit with some limitations. The number of charts displaying system metrics is limited, but you
+can use any of Netdata's [external plugins](../../../collectors/plugins.d/README.md) to monitor any services you might
+have installed on your macOS system. You could also use a macOS system as the master node in a [streaming
+configuration](../../../streaming/README.md).
 
-## With Homebrew
+We recommend installing Netdata with the community-created and -maintained [**Homebrew
+package**](#install-netdata-with-the-homebrew-package). 
 
-You can either install Netdata with [Homebrew](https://brew.sh/)
+-   [Install Netdata via the Homebrew package](#install-netdata-with-the-homebrew-package)
+-   [Install Netdata from source](#install-netdata-from-source)
+
+## Install Netdata with the Homebrew package
+
+If you don't have [Homebrew](https://brew.sh/) installed already, begin with their installation script:
+
+```bash
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+```
+
+Next, you can use Homebrew's package, which installs Netdata all its dependencies in a single step:
 
 ```sh
 brew install netdata
 ```
 
-## From source
+> Homebrew will place your Netdata configuration directory at `/usr/local/etc/netdata/`. Use the `edit-config` script
+> and the files in this directory to configure Netdata. For reference, you can find stock configuration files at
+> `/usr/local/Cellar/netdata/{NETDATA_VERSION}/lib/netdata/conf.d/`.
 
-or from source:
+Skip on ahead to the [What's next?](#whats-next) section to find links to helpful post-installation guides.
 
-```sh
-# install Xcode Command Line Tools
+## Install Netdata from source
+
+We don't recommend installing Netdata from source on macOS, as it can be difficult to configure and install dependencies
+manually.
+
+First open your terminal of choice and install the Xcode development packages.
+
+```bash
 xcode-select --install
 ```
 
-click `Install` in the software update popup window, then
+Click **Install** on the Software Update popup window that appears. Then, use the same terminal session to use Homebrew
+to install some of Netdata's prerequisites.
 
-```sh
-# install HomeBrew package manager
-/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+```bash
+brew install ossp-uuid autoconf automake pkg-config libuv lz4 json-c openssl@1.1
+```
+
+If you want to use the [database engine](../../../database/engine/README.md) to store your metrics, you need to download
+and install the [Judy library](https://sourceforge.net/projects/judy/) before proceeding compiling Netdata.
 
-# install required packages
-brew install ossp-uuid autoconf automake pkg-config
+Next, download Netdata from our GitHub repository:
 
-# download Netdata
+```bash
 git clone https://github.com/netdata/netdata.git --depth=100
```

Comment:
This line can be updated to use `--depth=1` instead of `--depth=100`. An install doesn't need the last 100 commits, only the most recent one.

Alternatively, if we think we need some history for some reason, just drop the `--depth` option altogether.

Reply:
i recall we made it 100 for a reason, it is related with google analytics somehow. We had some problems with it with 1.

Reply:
Fascinating! I'll do some digging to see what I can find about this.

Reply:
Here's the PR where it changed: https://github.com/netdata/netdata/pull/5119. I guess it had to do with moving the doc from `installer/README.md` to `packaging/installer/README.md`? I'm not very sure, to be honest.

My gut says keep it as-is for now.

Reply:
Yes, it was versioining problem, there were no version with depth=1 or something

Reply:
If this is actually related to analytics beyond ensuring versions are sane, I'll be amazed.

Digging a bit further, it looks like we need the history for `git describe` (which we use during the build) to work correctly. In this case, this is a classic example of an arbitrary magic number that doesn't actually fix anything, and should just be removed, because the case this is supposed to prevent from happening (no recent tags in the local history) can still happen (especially considering that users may set `depth` to whatever the hell they want anyway).

Reply:
Okay, I'll follow your lead on this. If you think it should be cut, I'll cut it!

Reply:
Posting one final comment here so that there's a proper record of my reasoning on this so that people don't have to go digging to find it in the future:

When building out of a git clone of the source, we use `git describe` to determine the version string. `git describe` looks for the most recent tag in the history of the current branch to construct the first part of the version, and then uses the commit hash as the second part if the current commit isn't the one with the discovered tag. In the event that there is no tag (as is usually the case when doing a shallow clone with a short history), it just uses the commit hash as the version, which is not what we want because it breaks assumptions made by many tools that try to parse version numbers.

The original change that bumped this to 100 was a prophylactic to mitigate this, but it's not really a correct fix because it's perfectly reasonable that the most recent tag won't be in the last 100 commits (in fact, I'm pretty sure it isn't as of the current master branch as I'm writing this), so we can still have the same type of failure here.

Realistically, a shallow clone is just an efficiency boost for end users and nothing else, but even a deep clone of `master` isn't particularly bad (we haven't even hit 100MB yet for the clone process), so there's not a huge benefit to doing a shallow clone unless you're on a super slow connection or have to pay ridiculous premiums for data usage.

Given all of this, there's not much point to having a `--depth` option here. Most users won't care about it, and the few that do should be savvy enough to use `--depth` appropriately.